lib/logflare_web/controllers/source_controller.ex

```diff
@@ -222,47 +222,60 @@ defmodule LogflareWeb.SourceController do
   end
 
   def test_alerts(conn, %{"id" => source_id}) do
-    source = Sources.get_by_and_preload(id: source_id)
-
-    case WebhookNotificationServer.test_post(source) do
-      {:ok, %Tesla.Env{} = _response} ->
-        conn
-        |> put_flash(:info, "Webhook test successful!")
-        |> redirect(to: Routes.source_path(conn, :edit, source.id))
-
-      {:error, %Tesla.Env{} = response} ->
-        conn
-        |> put_flash(:error, "Webhook test failed! Response status code was #{response.status}.")
-        |> redirect(to: Routes.source_path(conn, :edit, source.id))
-
-      {:error, response} ->
+    with source = %Source{} <- Sources.get_by_and_preload(id: source_id) do
+      case WebhookNotificationServer.test_post(source) do
+        {:ok, %Tesla.Env{} = _response} ->
+          conn
+          |> put_flash(:info, "Webhook test successful!")
+          |> redirect(to: Routes.source_path(conn, :edit, source.id))
+
+        {:error, %Tesla.Env{} = response} ->
```

Comment:
We can put these unhappy paths in the else as well

Reply:
Done in last refactor 🙇